test/widgets/lightbox_test.dart

```diff
@@ -289,6 +308,84 @@ void main() {
     //   https://github.com/zulip/zulip-flutter/pull/833#issuecomment-2251782337
   });
 
+  group('IconButton Download Test', () {
+    Future<void> buildTestWidget(WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Builder(
+              builder: (context) {
+                return IconButton(
+                  icon: const Icon(Icons.download_rounded),
+                  onPressed: () async {
+                    // Simulate download logic
+                    final response = await fakeHttpClient.get(Uri.parse('https://example.com/image.png'));
+
+                    if (response.statusCode == 200) {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        const SnackBar(content: Text("Image downloaded successfully")),
+                      );
+                    } else {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        const SnackBar(content: Text("Failed to download image")),
+                      );
+                    }
+                  },
+                );
+              },
+            ),
+          ),
+        ),
+      );
+    }
+
+    testWidgets('should display download button and trigger download', (tester) async {
+      fakeHttpClient = FakeHttpClient(
+        onRequest: (request) {
+          return http.Response('Image Data', 200);
+        },
+      );
+
+      await buildTestWidget(tester);
+
+      // Verify download icon exists
+      final downloadIcon = find.byIcon(Icons.download_rounded);
+      expect(downloadIcon, findsOneWidget);
```

Comment:
We should use `check` in our tests. See https://github.com/zulip/zulip-flutter?tab=readme-ov-file#check-vs-expect